integration/spark2/src/main/spark2.1/org/apache/spark/sql/catalyst/catalog/HiveTableRelation.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.catalog
+
+import com.google.common.base.Objects
+import org.apache.spark.sql.catalyst.analysis.MultiInstanceRelation
+import org.apache.spark.sql.catalyst.expressions.AttributeReference
+import org.apache.spark.sql.catalyst.plans.logical.{LeafNode, LogicalPlan, Statistics}
+import org.apache.spark.sql.internal.SQLConf
+
+/**
+  * A `LogicalPlan` that represents a hive table.
+  *
+  * TODO: remove this after we completely make hive as a data source.
+  */
+case class HiveTableRelation(
```

Comment:
I don't see the reason why we need this class. Please try to avoid adding a class unnecessarly

Reply:
@ravipesala 
Spark2.1 didn't have HiveTableRelation, but we use it in mv, such as LogicalPlanSignatureGenerator.scala

Reply:
There also you can consider the way it is done in `CarbonFileMetastore#lookupRelation` , no need to have sperate class.

Reply:
HiveTableRelation.class is used in several MV code, so  I think add a class is better than reflection in CarbonFileMetastore#lookupRelation

Reply:
You are creating a class in the same package with the same name that means you are overriding the original class. It would behave weirdly if this class is loaded in place of the original class.  
Better don't create the class with same name and override the behaviour. You can create class and handle the unapply the mv package.

Reply:
1. We can't move ```HiveTableRelation``` to mv package, because we need to keep consistent with the spark 2.2&2.3 code 
2. ```HiveTableRelation``` only be put in spark 2.1 adapt code, it will not affect existing features in spark 2.1, because spark 2.1 source code didn't hive this class
So, I still suggest use it in spark 2.1 adapt code